### PR TITLE
propagate outputs to shareholders on enclaves

### PR DESCRIPTION
**Pull request by koenvanderveen**

### CODE REVIEW

File: packages/syft/src/syft/client/enclave_client.py [hunk 1]
```diff
@@ -42,6 +42,7 @@ class EnclaveClient(SyftClient):
 
     @property
     def code(self) -> Optional[APIModule]:
+        self
```
Comment (shubham3121) [inline packages/syft/src/syft/client/enclave_client.py]: this needs to be removed ?

File: packages/syft/src/syft/service/action/action_service.py [hunk 2]
```diff
@@ -211,8 +213,18 @@ def _user_code_execute(
             syft_object=result_action_object,
             has_result_read_permission=True,
         )
+
         if set_result.is_err():
             return set_result.err()
+
+        if len(code_item.output_policy.output_readers) > 0:
```
Comment (shubham3121) [inline packages/syft/src/syft/service/action/action_service.py:220]: 👍🏾

File: packages/syft/src/syft/service/code/user_code.py [hunk 3]
```diff
@@ -507,16 +558,20 @@ def debox_asset(arg: Any) -> Any:
...
         input_policy=ExactMatch(*args, **kwargs),
         output_policy=SingleExecutionExactOutput(),
+        share_results_with_owners=share_results_with_owners,
     )
 
 
 def syft_function(
     input_policy: Union[InputPolicy, UID],
     output_policy: Optional[Union[OutputPolicy, UID]] = None,
+    share_results_with_owners=False,
```
Comment (shubham3121) [inline packages/syft/src/syft/service/code/user_code.py:574]: Should we make it True by default because this is something that DS controls ?

**Review by shubham3121 — APPROVED** [review]
Excellent work @koenvanderveen !! 
This makes the code much more clean and readable 💯 🎉

File: packages/syft/src/syft/service/code/user_code.py (hunk 3 above)
Comment (koenvanderveen) [reply packages/syft/src/syft/service/code/user_code.py:574]: irina suggested False, I agree. In theory the output still could be data that is sensitive for the data owners so I would not have it on by default

File: packages/syft/src/syft/client/enclave_client.py (hunk 1 above)
Comment (koenvanderveen) [reply packages/syft/src/syft/client/enclave_client.py]: yes, done

**Pull request merged**